### ISSUE TITLE
Error with recent Node versions (>=17) 

### ISSUE THREAD

**Issue by FridIsar** [Migracode-Barcelona/react-hotel#1]
Error when using a recent enough Node version (>=17) 

Node.js v20.3.0

[Temporary fix](https://stackoverflow.com/questions/69692842/error-message-error0308010cdigital-envelope-routinesunsupported)

There is "502 Bad Gateway" error showing up. The following is the error message:
```
[12:05](https://migracodebarcelona.slack.com/archives/CMDSP2CQ2/p1690452307670279)
Starting the development server...
Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:69:19)
    at Object.createHash (node:crypto:138:10)
    at module.exports (/workspaces/hotel-keithtou/node_modules/webpack/lib/util/createHash.js:90:53)
    at NormalModule._initBuildHash (/workspaces/hotel-keithtou/node_modules/webpack/lib/NormalModule.js:386:16)
    at handleParseError (/workspaces/hotel-keithtou/node_modules/webpack/lib/NormalModule.js:434:10)
    at /workspaces/hotel-keithtou/node_modules/webpack/lib/NormalModule.js:466:5
    at /workspaces/hotel-keithtou/node_modules/webpack/lib/NormalModule.js:327:12
    at /workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:373:3
    at iterateNormalLoaders (/workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:214:10)
    at iterateNormalLoaders (/workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:221:10)
    at /workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:236:3
    at runSyncOrAsync (/workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:130:11)
    at iterateNormalLoaders (/workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:232:2)
    at Array.<anonymous> (/workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:205:4)
    at Storage.finished (/workspaces/hotel-keithtou/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:55:16)
    at /workspaces/hotel-keithtou/node_modules/enhanced-resolve/lib/CachedInputFileSystem.js:91:9
/workspaces/hotel-keithtou/node_modules/react-scripts/scripts/start.js:19
  throw err;
  ^
Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:69:19)
    at Object.createHash (node:crypto:138:10)
    at module.exports (/workspaces/hotel-keithtou/node_modules/webpack/lib/util/createHash.js:90:53)
    at NormalModule._initBuildHash (/workspaces/hotel-keithtou/node_modules/webpack/lib/NormalModule.js:386:16)
    at /workspaces/hotel-keithtou/node_modules/webpack/lib/NormalModule.js:418:10
    at /workspaces/hotel-keithtou/node_modules/webpack/lib/NormalModule.js:293:13
    at /workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:367:11
    at /workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:233:18
    at context.callback (/workspaces/hotel-keithtou/node_modules/loader-runner/lib/LoaderRunner.js:111:13)
    at /workspaces/hotel-keithtou/node_modules/babel-loader/lib/index.js:51:103 {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}
```

**Comment by FridIsar** [comment]
@unflxw I can't replicate the issue so I don't know what the definite fix. Is it changing the version of webpack inside package-lock.json?

**Comment by unflxw** [comment]
I was familiar with this issue beforehand, back from when we upgraded to Node 17 at work, which is why I suggested others might encounter this issue as well. But now that I take a closer look at it, it's unclear to me how the student encountered this issue, and I cannot reproduce it either.

The issue on webpack's side was fixed with [this PR](https://github.com/webpack/webpack/pull/14584), which was [released in version `5.61.0`](https://github.com/webpack/webpack/releases/tag/v5.61.0). The `package-lock.json` in this project is [already set to a higher version (`5.75.0`)](https://github.com/Migracode-Barcelona/react-hotel/blob/master/package-lock.json#L16004-L16005). So I do not think that webpack needs to be upgraded further.

Is it possible that the affected student was somehow using a very old clone of this repository?

**Comment by FridIsar** [comment]
I have tried replicating it with the student's repository prior and after he updated the package-lock.json and both worked for me, so I am not sure what could have caused the issue on his side 🤔 maybe a specific version of Node causes that issue, or some other package combination. I'll close the issue for now, let's see if it happens again in the future. Thanks for looking into it!
